src/lib/utils/scales-utils.js

```diff
@@ -615,3 +622,28 @@ export function getMissingScaleProps(props, data, attributes) {
   });
   return result;
 }
+
+/**
+ * Return a d3 scale that returns the literal value that was given to it
+ * @returns {function} literal scale.
+ */
+export function literalScale() {
+  function scale(d) {
+    return d;
+  }
+
+  scale.domain = function domain() {
```

Comment:
Might want to shorten all these to `() => scale;`, or DRY this with one function, since all the functions are performing exactly the same operation.